### PR TITLE
auto_backup: allow to change the format of backup

### PR DESCRIPTION
**Pull request by foutoucour**
Add the option to do the backup without the filestore.
We found ourselves the need for hourly backups.

### CODE REVIEW

File: auto_backup/models/db_backup.py [hunk 1]
```diff
@@ -151,21 +158,23 @@ def action_backup(self):
...
                 successful |= rec
 
         # Ensure a local backup exists if we are going to write it remotely
         sftp = self.filtered(lambda r: r.method == "sftp")
         if sftp:
-            if backup:
-                cached = open(backup)
-            else:
-                cached = db.dump_db(self.env.cr.dbname, None)
+            for rec in sftp:
```
Comment (ddufresne) [inline auto_backup/models/db_backup.py:174]: @foutoucour the if statement can be removed since you loop on the for.

**Review by ddufresne — CHANGES_REQUESTED** [review]
Minor issue

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+version: '3'
```
Comment (astirpe) [inline docker-compose.yml]: Could you remove file `docker-compose.yml` from this PR?

File: auto_backup/models/db_backup.py [hunk 3]
```diff
@@ -83,6 +83,13 @@ class DbBackup(models.Model):
              "read permissions for that file.",
     )
 
+    backup_format = fields.Selection(
+        [("dump.zip", "zip (includes filestore)"), ("dump", "pg_dump custom format (without filestore)")],
```
Comment (astirpe) [inline auto_backup/models/db_backup.py]: Shouldn't be "`zip`" here? I'm wondering how this functionality could work fine with "`dump.zip`" passed to `dump_db()` :thinking:

**Review by astirpe — CHANGES_REQUESTED** [review]

File: auto_backup/models/db_backup.py (hunk 3 above)
Comment (foutoucour) [reply auto_backup/models/db_backup.py]: True!

Comment (foutoucour) [reply auto_backup/models/db_backup.py]: @astirpe fixed!

File: auto_backup/tests/test_db_backup.py [hunk 4]
```diff
@@ -175,18 +175,6 @@ def test_action_backup_sftp_remote_open(self):
                     'wb'
                 )
 
-    def test_action_backup_sftp_remote_open(self):
```
Comment (astirpe) [inline auto_backup/tests/test_db_backup.py:178]: Why this test was removed?

File: auto_backup/models/db_backup.py [hunk 5]
```diff
@@ -83,6 +83,16 @@ class DbBackup(models.Model):
              "read permissions for that file.",
     )
 
+    backup_format = fields.Selection(
+        [
+            ("zip", "zip (includes filestore)"),
+            ("dump", "pg_dump custom format (without filestore)")
+        ],
+        string="Backup Format",
```
Comment (astirpe) [inline auto_backup/models/db_backup.py]: No need to write `string="Backup Format",`

File: auto_backup/tests/test_db_backup.py (hunk 4 above)
Comment (foutoucour) [reply auto_backup/tests/test_db_backup.py:178]: Hello, 
I found that it is the same test than the previous test

File: auto_backup/models/db_backup.py (hunk 5 above)
Comment (foutoucour) [reply auto_backup/models/db_backup.py]: Question of style. I prefer explicit declaration that implicit declaration.
Has OCA guide on this that I didn't see?

Comment (astirpe) [reply auto_backup/models/db_backup.py]: See https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#fields

File: auto_backup/tests/test_db_backup.py (hunk 4 above)
Comment (astirpe) [reply auto_backup/tests/test_db_backup.py:178]: Your're right

File: auto_backup/models/db_backup.py (hunk 5 above)
Comment (foutoucour) [reply auto_backup/models/db_backup.py]: 👍

**Comment by foutoucour** [comment]
@astirpe could you approve the PR? Thank you.

**Review by astirpe — APPROVED** [review]
Code review ok

**Review by ddufresne — APPROVED** [review]

**Pull request merged**